### PR TITLE
Document new `DONT_TRAVERSER_CURRENT_AND_CHILDREN` constant

### PR DESCRIPTION
**Pull request by borNfreee**
This PR adds missed documentation for already merged https://github.com/nikic/PHP-Parser/pull/536

### CODE REVIEW

File: doc/component/Walking_the_AST.markdown [hunk 1]
```diff
@@ -281,6 +281,8 @@ special enterNode/leaveNode return values:
 
  * If *any* visitor returns `DONT_TRAVERSE_CHILDREN`, the children will be skipped for *all*
    visitors.
+* If *any* visitor returns `DONT_TRAVERSER_CURRENT_AND_CHILDREN`, the children will be skipped for *all*
```
Comment (nikic) [inline doc/component/Walking_the_AST.markdown]: nit: Indentation is off by one.

File: doc/2_Usage_of_basic_components.markdown [hunk 2]
```diff
@@ -341,6 +341,9 @@ case the current node is not changed.
 The `enterNode()` method can additionally return the value `NodeTraverser::DONT_TRAVERSE_CHILDREN`,
 which instructs the traverser to skip all children of the current node.
 
+The `enterNode()` method can additionally return the value `NodeTraverser::DONT_TRAVERSER_CURRENT_AND_CHILDREN`
+to skip both traversal of child nodes, and prevent subsequent visitors from visiting the current node.
```
Comment (nikic) [inline doc/2_Usage_of_basic_components.markdown]: The duplication of the sentence structure here sounds a bit awkward. I'd suggest combining with the previous paragraph into something like

> The `enterNode()` method can additionally return the value `NodeTraverser::DONT_TRAVERSE_CHILDREN`, which instructs the traverser to skip all children of the current node. To furthermore prevent subsequent visitors from visiting the current node, `NodeTraverser::DONT_TRAVERSE_CURRENT_AND_CHILDREN` can be used instead.

**Comment by nikic** [comment]
Thanks!

**Comment by borNfreee** [comment]
thank you

**Comment by phil-davis** [comment]
@borNfreee can you please edit the title of this PR to correct the name of the constant:
s/DONT_TRAVERSER_CURRENT_AND_CHILDREN/DONT_TRAVERSE_CURRENT_AND_CHILDREN/

That will reduce any future confusion about the exact constant name.

Unfortunately the mis-spelling is in the commit title also - hopefully nobody does cut-paste from there instead of reading the doc!

**Comment by borNfreee** [comment]
@phil-davis done, thank you

**Pull request merged**